medicines/api/src/document.rs

```diff
@@ -55,31 +39,235 @@ pagination! {Documents, DocumentEdge, Document}
 
 pub async fn get_documents(
     client: &impl Search,
-    search: Option<String>,
+    search: String,
     first: Option<i32>,
-    last: Option<i32>,
-    before: Option<String>,
     after: Option<String>,
-) -> Documents {
-    let placeholder_names: [&str; 1000] = ["A cool document"; 1000];
-    let edges = placeholder_names
+) -> Result<Documents, anyhow::Error> {
+    let result_count = first.unwrap_or(10);
+    let offset = match after {
+        Some(after) => {
+            let bytes = base64::decode(after)?;
+            let string = std::str::from_utf8(&bytes)?;
+            string.parse::<i32>()? + 1
+        }
+        None => 0,
+    };
+
+    let azure_result = client
+        .search_with_pagination(
+            &search,
+            search_client::AzurePagination {
+                result_count: result_count,
+                offset: offset,
+            },
+            true,
+        )
+        .await?;
+
+    let mut cursor = offset.clone();
+    let edges = azure_result
+        .search_results
         .iter()
-        // .take(first as usize)
-        .map(|&name| Document::name_only(name.to_string()))
-        .map(|document| DocumentEdge {
-            node: document,
-            cursor: "cursor".to_owned(),
+        .map(|search_result| Document::from(search_result.clone()))
+        .map(|document| {
+            let edge = DocumentEdge {
+                node: document,
+                cursor: base64::encode(cursor.to_string()),
+            };
+            cursor += 1;
```

Comment:
Just a suggestion but you could use [`.enumerate`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) to get the index of the document. e.g.

```rust
.enumerate()
.map(|(i, document| {
  DocumentEdge {
    node: document,
    cursor: base64::encode((offset + i).to_string()),
  }
})